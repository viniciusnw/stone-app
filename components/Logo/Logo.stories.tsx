import React from "react";
import { View } from "react-native";
import { Logo, StorybookWrapper } from "@Components";

export const Default = {};

export default {
  title: "Logo",
  component: () => (
    <StorybookWrapper>
      <View style={{ marginTop: 60 }}>
        <Logo />
      </View>
    </StorybookWrapper>
  ),
};
