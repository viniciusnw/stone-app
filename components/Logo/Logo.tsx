import React from 'react';
import styled from 'styled-components/native';
import { View, Text, StyleSheet } from 'react-native';

const StyledContainer = styled(View)`
  flex-direction: row;
`;

const StyledStone = styled(Text)`
  color: ${({ theme }) => theme.color.default};
  font-weight: 700;
  font-size: 32px;
`;

const StyledTon = styled(Text)`
  color: ${({ theme }) => theme.color.primary};
  font-weight: 700;
  font-size: 32px;
`;

const fontStyle = StyleSheet.create({
  FredokaOneRegular: {
    fontFamily: 'FredokaOne-Regular'
  },
});


const Logo = (props: any) => {
  return (
    <StyledContainer {...props}>
      <Text style={fontStyle.FredokaOneRegular}>
        <StyledStone>
          Stone
        </StyledStone>
        <StyledTon>
          Ton :)
        </StyledTon>
      </Text>
    </StyledContainer>
  )
};

export default Logo;
