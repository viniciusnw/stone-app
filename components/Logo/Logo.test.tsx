import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Logo from "./Logo";

jest.useFakeTimers();

test("Logo", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Logo />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
