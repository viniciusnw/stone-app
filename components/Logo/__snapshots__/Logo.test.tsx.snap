// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "fontFamily": "FredokaOne-Regular",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(0,168,104)",
          "fontSize": 32,
          "fontWeight": "700",
        }
      }
    >
      Stone
    </Text>
    <Text
      style={
        {
          "color": "rgb(10,245,61)",
          "fontSize": 32,
          "fontWeight": "700",
        }
      }
    >
      Ton :)
    </Text>
  </Text>
</View>
`;
